### DISCUSSION TITLE
Sideload app from App Store with SIP disabled

### DISCUSSION THREAD

**Discussion by malash** [PlayCover/PlayCover#273] [General]
Hi folks, I love this project because I usually disabled SIP for some software development reasons and `PlayCover` re-enabled the ability to run iOS app on my M1 mac.

In some cases users need to disable SIP, like developing kernel extensions, using [3rd-party tools](https://github.com/rofl0r/proxychains-ng) or just want to fix `PlayCover` login issue, after that all apps downloaded from App Store will stop running anymore.

|              | Apps from App Store | Decrypted IPAs                        |
| ------------ | ------------------- | ------------------------------------- |
| SIP enabled  | Yes                 | Partial Yes ( with some login issue ) |
| SIP disabled | No                  | Yes                                   |

Although you can download decryped IPAs and run them by`PlayCover`, only a few part of IPAs can be found on internet. Another way to get IPAs is using a Jailbreak iOS device, it still has some thresholds for normal users.

So I'm wondering is it possiable to:

1. Get decrypted IPAs from macOS App Store ( since it's more easier to _Jailbreak_ a macOS device than iOS, right ? )

or

2. Sideload `.app` downloaded from App Store directly by `PlayCover`.

Thanks.

**Comment by JoseMoreville** [comment]
Sadly, decrypting on mac isn't possible after 11.3 and we currently have no method to sideload `.app` from appstore without having to decrypt them. If you have a solution you're welcome to share it

**Comment by malash** [comment]
I see, it seems the key is to decrypt the app.

I find some tools that said they can decrypt iOS apps on M1 mac.

https://github.com/paradiseduo/appdecrypt
https://github.com/Lakr233/Decrypter ( video: https://www.youtube.com/watch?v=UX_5fPp8t1M )
https://github.com/meme/apple-tools/tree/master/foulplay

But saidly non of them works on my M1 Max with macOS 12.5.1
I have no idea whether is it a limitation from macOS >= 11.3. My M1 Max device can only downgrade to macOS 12.0 so I have no way to verify it.
